bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -210,11 +162,35 @@ public ObjectId(final Date date, final int machineIdentifier, final short proces
      * @param processIdentifier the process identifier
      * @param counter           the counter
      * @throws IllegalArgumentException if the high order byte of machineIdentifier or counter is not zero
+     * @deprecated Use {@link #ObjectId(int, int)} instead
      */
+    @Deprecated
     public ObjectId(final int timestamp, final int machineIdentifier, final short processIdentifier, final int counter) {
         this(timestamp, machineIdentifier, processIdentifier, counter, true);
     }
 
+    /**
+     * Creates an ObjectId using the given time, machine identifier, process identifier, and counter.
+     *
+     * @param timestamp         the time in seconds
+     * @param counter           the counter
+     * @throws IllegalArgumentException if the high order byte of counter is not zero
+     */
+    public ObjectId(final int timestamp, final int counter) {
+        this(timestamp, counter, true);
+    }
+
+    private ObjectId(final int timestamp, final int counter, final boolean checkCounter) {
+        if (checkCounter && ((counter & 0xff000000) != 0)) {
+            throw new IllegalArgumentException("The counter must be between 0 and 16777215 (it must fit in three bytes).");
+        }
+        this.timestamp = timestamp;
+
+        Random random = new Random();
```

Comment:
Spec says:

> A 5-byte field consisting of a random value generated once per process.

So this should be done in a static initializer.  And it should use the `SecureRandom` class rather than the `Random` class

Reply:
I have added the static initializer for the random value integer fields.